### PR TITLE
fix #5506: Make processId and cwd in TerminalWidget return promise when throw error

### LINKED ISSUE #5506: [terminal] 'terminal is not started'
### Description
When I open the terminal, the console will report this error.

### Reproduction Steps
In my case, I started theia from the image, and there is no problem with the image running locally, but when I run it remotely, I get an error, which I think may be related to slower network interaction.

**Diagnostics:** 
```console
38.bundle.js:461 Uncaught (in promise) Error: terminal is not started
    at TerminalWidgetImpl.get [as processId] (38.bundle.js:461)
    at updateProcessId (90.bundle.js:10063)
    at TerminalServiceMainImpl.<anonymous> (90.bundle.js:10064)
```
After the terminal is created by the WidgetManager, the trackTerminal method of the TerminalServiceMain in the plugin-ext is triggered, but the processId of the current TerminalWidgetImpl does not guarantee that the start method call is completed when used, so the correct processId cannot be obtained.

### PR DESCRIPTION
**Pull request by BroKun**
Add the ready interface to ensure that the correct processId can be obtained.

Signed-off-by: Brokun <brokun0128@gmail.com>

### CODE REVIEW

**Comment by BroKun** [comment]
Currently, it works for me. 
I think we can look at whether to avoid multiple calls to resolve and whether to update 'ready' when attach.

File: packages/terminal/src/browser/terminal-widget-impl.ts [hunk 1]
```diff
@@ -203,22 +204,30 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         return this.term;
     }
 
+    get ready(): Promise<void> {
```
Comment (akosyakov) [inline packages/terminal/src/browser/terminal-widget-impl.ts]: please don't expose it for now, keep protected

**Comment by akosyakov** [comment]
@BroKun How do you test it? Code-wise it looks good.

**Comment by BroKun** [comment]
> @BroKun How do you test it? Code-wise it looks good.

I modified the solution to this problem because I noticed that onDidOpen can fix the exceptions caused by the call sequence. However, since processId and cwd cannot be captured by reject when an exception is thrown, it causes a problem.

As for testing, I feel that the current solution does not require special testing. The problem scenario can be observed by adding a wait time before the start method.

**Comment by akosyakov** [comment]
> As for testing, I feel that the current solution does not require special testing. The problem scenario can be observed by adding a wait time before the start method.

It is better to be safe and double check.

**Pull request merged**